static/src/javascripts-legacy/bootstraps/enhanced/recipe-article.js

```diff
@@ -0,0 +1,118 @@
+define([
+    'common/utils/$',
+    'common/utils/scroller',
+    'bean',
+    'lodash/functions/debounce'
+], function (
+    $,
+    scroller,
+    bean,
+    debounce
+) {
+
+function init() {
+    var recipe = $('.recipe__article--structured');
+    var firstRecipe = $('.js-recipe__article--structured .js-recipe__article--structured-headline h1')
+    var nextWrapper = $('.js-recipe__article--next-recipe');
+    var nextRecipeTitle = $('.js-recipe__article--next-title');
+    var nextRecipeText = $('.js-recipe__article--next-text');
+    var nextRecipeKicker = $('.js-kicker');
+    var nextButton = $('.js-recipe__article--next-button');
+    var readMoreButton = $('.js-recipe__article--read-more');
+    var readMoreButtonText = $('.read-more__text');
+    var displayClass = 'recipe__image__wrapper--is-displayed';
+
+    function setKicker() {
+        var nextRecipe = getNextRecipe();
+
+        if (nextRecipe && nextRecipe.length > 0) {
+            nextButton.removeClass('top');
+            nextRecipeTitle.html(nextRecipe.html());
+            nextRecipeKicker.html('Next recipe ');
+        } else {
+            nextButton.addClass('top');
+            nextRecipeTitle.html(firstRecipe.html());
+            nextRecipeKicker.html('First recipe ');
+        }
+    }
+
+    function setMainImage(recipeInView){
+        var visibleRecipe = recipeInView.dataset['recipeIndex'];
+        var displayedImage = $('.' + displayClass);
+        var relatedImage = $('.' + visibleRecipe);
+        if (relatedImage.get(0) !== displayedImage.get(0)) {
+            displayedImage.removeClass(displayClass);
+            relatedImage.addClass(displayClass);
+        }
+    }
+
+    function getNextRecipe() {
+        return $('.js-recipe__article--structured.inview + .js-recipe__article--structured .js-recipe__article--structured-headline h1');
+    }
+
+    function getIntOfRecipeInView() {
+        var inView;
+        for (var i = 0; i < recipe.length; i++) {
+            if (isInView(i)) { inView = i; break; }
+        }
+        return inView >= 0 ? inView : -1
+    }
+
+    function isInView(i) {
```

Comment:
Lib looks good but I think it could return true for more than one recipe at at time. This function should return false if the recipe is in the bottom half of the page - so`isInView` is misleading name and will change.